### PR TITLE
[AIRFLOW-6561] Add possibility to specify default resources for airfl…

### PR DESCRIPTION
**Pull request by stijndehaes**
…ow k8s workers

---
Issue link: [AIRFLOW-6561](https://issues.apache.org/jira/browse/AIRFLOW-6561)

Make sure to mark the boxes below before creating PR: [x]

- [x] Description above provides context of the change
- [x] Commit message/PR title starts with `[AIRFLOW-NNNN]`. AIRFLOW-NNNN = JIRA ID<sup>*</sup>
- [x] Unit tests coverage for changes (not needed for documentation changes)
- [x] Commits follow "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)"
- [x] Relevant documentation is updated including usage instructions.
- [x] I will engage committers as explained in [Contribution Workflow Example](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#contribution-workflow-example).

<sup>*</sup> For document-only changes commit message can start with `[AIRFLOW-XXXX]`.

---
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/master/UPDATING.md).
Read the [Pull Request Guidelines](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#pull-request-guidelines) for more information.

### CODE REVIEW

File: airflow/executors/kubernetes_executor.py [hunk 1]
```diff
@@ -110,6 +110,12 @@ def __init__(self):  # pylint: disable=too-many-statements
         self.worker_run_as_user = self._get_security_context_val('run_as_user')
         self.worker_fs_group = self._get_security_context_val('fs_group')
 
+        kube_worker_resources = conf.get(self.kubernetes_section, 'worker_resources')
+        if kube_worker_resources:
+            self.worker_resources = json.loads(kube_worker_resources)
+        else:
+            self.worker_resources = None
```
Comment (davlum) [inline airflow/executors/kubernetes_executor.py]:
```suggestion
        kube_worker_resources = conf.get(self.kubernetes_section, 'worker_resources', fallback=None)
```

**Comment by davlum** [comment]
The JSON you are passing will not turn into a K8S resource object. I would suggest deserializing into a `kubernetes.client.models.V1ResourceRequirements` object in the `WorkerConfiguration` class.

Hypothetically you could pass JSON which conforms to the K8S api;
`{"limits": {"cpu":"foo", "memory":"bar"}, "requests": {"cpu":"foo", "memory":"bar"}}`, but I think deserializing to the object is the better choice.

**Comment by potiuk** [comment]
Please rebase to latest master @stijndehaes. We had some six bundling problem in Travis but it is now solved

**Comment by stijndehaes** [comment]
@potiuk rebased the branch to master

**Review by potiuk — APPROVED** [review]
this looks good to me . @dimberman  - maybe you can also take a look at that? Do you think it is good to define this way?

**Comment by dimberman** [comment]
Hi @potiuk I think this looks great :). Good idea to have the defaults here and then users can use the executor_config for individual tasks. Thank you @stijndehaes

**Pull request merged**